cmake/modules/SwiftComponents.cmake

```diff
@@ -122,3 +122,29 @@ function(swift_install_in_component component)
     install(${ARGN})
   endif()
 endfunction()
+
+function(swift_install_symlink_component component)
+  cmake_parse_arguments(
+      ARG # prefix
+      "" "LINK_NAME;TARGET;DESTINATION" "" ${ARGN})
```

Comment:
Better yet, we should suggest that syntax to the CMake developers and see if they'll run with it.

That said, I generally think that the bar for providing a wrapper around built-in CMake functionality should be pretty high. CMake has excellent online documentation, and providing custom wrappers for built-in functionality makes the code less approachable from someone outside the project. I do like the syntax you're suggesting, but IMO the benefit isn't substantial enough to break from convention.